Assets/Scripts/UI/Pickup Pup/Core/PPUIController.cs

```diff
@@ -41,17 +57,46 @@ public void LoadHome()
 		sceneController.LoadHome();
 	}
 
+    public void LoadShelter()
+    {
+        sceneController.LoadShelter();
+    }
+
+    public void LoadShop()
+    {
+        sceneController.LoadShop();
+    }
+
 	void handlePPDogEvent(PPEvent gameEvent, Dog dog)
 	{
 		if(gameEvent == PPEvent.ClickDogSlot)
 		{
-			handleDogSlotClicked(dog);
+            selectedDog = dog;
+			handleDogSlotClicked(selectedDog);
 		}
 	}
 
 	void handleDogSlotClicked(Dog dog)
 	{
-		// TODO: Insert universal dog slot handle code here
+        // TODO: Insert universal dog slot handle code here
+        if (dogProfileObject != null)
```

Comment:
For any child of the Unity Component class (MonoBehaviour, Sprite, Collider, etc), you don't need to use  `== null` in a null check. You can just write:
```
if (dogProfileObject)
{
     ... 
}
```

Obviously not a required convention, but I feel like it looks cleaner